### ISSUE TITLE
Random display of "raw code" instead of crypto values

### ISSUE THREAD

**Issue by alexis-all** [chrislennon/Crypto-Touchbar-App#71]
I am using the Crypto Touchbar App tool for 2 hours now and I have seen several times this kind of bug instead of my favorite crypto-values (see picture)

![crypto](https://user-images.githubusercontent.com/35690037/35219722-0a7d584a-ff74-11e7-9b10-676b2353bc6f.jpg)

On this picture it appears instead of BTC value but it appears also on others crypto.
The raw code appears until the next auto-refresh (set in my case at 5 seconds)

**Comment by chrislennon** [comment]
@alexis-all thanks for the bug report!

Which output format (advanced options) did you select?

I pushed a fix for something similar a couple of hours ago #69

**Comment by alexis-all** [comment]
you're welcome ! Here is the code (for BTC but same for other crypto I've selected) :

```
import urllib2,json,sys

data = urllib2.urlopen("https://min-api.cryptocompare.com/data/pricemultifull?fsyms=BTC&tsyms=USD")
obj=json.load(data)
current = "{:0.1f}".format(obj["RAW"]["BTC"]["USD"]["PRICE"])
opening = obj["RAW"]["BTC"]["USD"]["OPEN24HOUR"]
high = "{:0.1f}".format(obj["RAW"]["BTC"]["USD"]["HIGHDAY"])
low = "{:0.1f}".format(obj["RAW"]["BTC"]["USD"]["LOWDAY"])
raw_current = float(obj["RAW"]["BTC"]["USD"]["PRICE"])
raw_opening = float(obj["RAW"]["BTC"]["USD"]["OPEN24HOUR"])
raw_high = float(obj["RAW"]["BTC"]["USD"]["HIGHDAY"])
raw_low = float(obj["RAW"]["BTC"]["USD"]["LOWDAY"])

print ("$" + current + " (" + str(round (((raw_current - raw_opening) / raw_current) * 100)) + "%)")
```

Actually I've only set my parameters from the website then imported the .json to BTT :
- BTC, ETH, XRP
- in USD ($)
- Live price
- Limit decimals to 1
- Change % based on opening (+10%)
- Auto refresh set : 5 seconds

Hope it helps

**Comment by chrislennon** [comment]
I see the code has my fixes from earlier so its not that.

The above runs locally too with no issue. Any chance you can open BTT, open the malfunctioning widget, run the script manually and send the error back?

I'm running the same settings on my touchbar now incase it is something strange being responded in the API (occasionally).

**Comment by alexis-all** [comment]
After several times I managed to get the error by running the script manually in BTT, here it is :

```
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 431, in open
    response = self._open(req, data)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 449, in _open
    '_open', req)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 1240, in https_open
    context=self._context)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 1197, in do_open
    raise URLError(err)
urllib2.URLError: <urlopen error [Errno 60] Operation timed out>
```

**Comment by chrislennon** [comment]
ah ha! so its a python timeout error (I have raised an issue #72 to handle the error output better).

So this is an _interesting_ one, some steps to debug:
- I assume you have network access?
- Can you reach the API (checking in a browser will be fine) [BTC-USD](https://min-api.cryptocompare.com/data/pricemultifull?fsyms=BTC&tsyms=USD)
- You could also save the code to a file and execute it with `python test.py` to see if the issue remains the same.
- If its happening only _occasionally_ you could potentially reduce the refresh interval, perhaps a new request is being started before the previous has completed.

**Comment by alexis-all** [comment]
> If its happening only occasionally you could potentially reduce the refresh interval, perhaps a new request is being started before the previous has completed.

that's what I thought of just after responding. I also think it comes from my refresh time which is too short, my network debit rate is quite slow at the moment. I just set the refresh time to 30 seconds and it seems OK... 

To answer : yes I have network access, and I can reach the api via Safari from your link.
I ran the code from python and get a correct value.

**Comment by chrislennon** [comment]
Alright, lets keep an eye on it and see if that resolves it 😃 

Thinking about it - and reading the [api docs](https://www.cryptocompare.com/api/#) (which state don't request more than once per 10 seconds ), maybe I should restrict it so the tool doesn't allow low refresh times (#73).

**Comment by chrislennon** [comment]
This issue has been closed as part of #76.

After moving to python away from applescript there were a few issues regarding formatting options. This PR should resolve the issues going forward, please let me know if it works and of course if the issue continues reopen this issue and we can debug further.
